src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs

```diff
@@ -332,6 +332,48 @@ static async ValueTask<int> FinishReadAsync(Task<int> readTask, byte[] localBuff
             }
         }
 
+        public async ValueTask ReadExactlyAsync(Memory<byte> buffer, CancellationToken cancellationToken = default) =>
+            _ = await ReadAtLeastAsyncCore(buffer, buffer.Length, throwOnEndOfStream: true, cancellationToken).ConfigureAwait(false);
+
+        public async ValueTask ReadExactlyAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken = default)
+        {
+            ValidateBufferArguments(buffer, offset, count);
+
+            _ = await ReadAtLeastAsyncCore(buffer.AsMemory(offset, count), count, throwOnEndOfStream: true, cancellationToken).ConfigureAwait(false);
+        }
+
+        public ValueTask<int> ReadAtLeastAsync(Memory<byte> buffer, int minimumBytes, bool throwOnEndOfStream = true, CancellationToken cancellationToken = default)
+        {
+            ValidateReadAtLeastArguments(buffer.Length, minimumBytes);
+
+            return ReadAtLeastAsyncCore(buffer, minimumBytes, throwOnEndOfStream: true, cancellationToken);
```

Comment:
```suggestion
            return ReadAtLeastAsyncCore(buffer, minimumBytes, throwOnEndOfStream, cancellationToken);
```

Reply:
Oops - copy and paste error. Thanks for the catch. I'm writing unit tests now, I'll make sure I add a test for this too.